psydac/core/bsplines.py

```diff
@@ -947,3 +947,110 @@ def basis_ders_on_irregular_grid(knots, degree, i_grid, cell_index, nders, norma
         assert out.shape == (nx, degree + 1, nders + 1) and out.dtype == np.dtype('float')
     basis_ders_on_irregular_grid_p(knots, degree, i_grid, cell_index, nders, normalization == 'M', out)
     return out
+
+#==============================================================================
+def _refinement_matrix_one_stage(t, p, knots):
+    """
+    computes the refinement matrix corresponding to the insertion of a given knot
+    """
```

Comment:
Please expand the docstring. What does this function return? Can you provide a reference for the algorithm?

Reply:
done in [690a80e](https://github.com/pyccel/psydac/pull/241/commits/690a80e3e4a4ec8d7e7a698a6ab38964c2c0b098)

Reply:
for the reference I don't have it, this was written by @ratnania

Reply:
Maybe it's The NURBS Book?

Reply:
done in [bdc2fe3](https://github.com/pyccel/psydac/pull/241/commits/bdc2fe3694702e1e589480177d0945cf5ef2872b)